headers/xdp/parsing_helpers.h

```diff
@@ -101,6 +102,39 @@ static __always_inline int parse_ethhdr(struct hdr_cursor *nh, void *data_end,
 	return h_proto; /* network-byte-order */
 }
 
+static __always_inline int skip_ip6hdrext(struct hdr_cursor *nh,
+					  void *data_end,
+					  __u8 currhdr)
+{
+	for (int i = 0; i < 8; ++i) {
+		struct ipv6_opt_hdr *hdr = nh->pos;
+
+		if (hdr + 1 > data_end)
+			return -1;
+
+		switch (currhdr) {
+		case IPPROTO_HOPOPTS:
+		case IPPROTO_DSTOPTS:
+		case IPPROTO_ROUTING:
+		case IPPROTO_MH:
+			nh->pos = (char *)hdr + (hdr->hdrlen + 1) * 8;
+			break;
+		case IPPROTO_AH:
+			nh->pos = (char *)hdr + (hdr->hdrlen + 2) * 4;
+			break;
+		case IPPROTO_FRAGMENT:
+			nh->pos = (char *)hdr + 8;
+			break;
+		default:
+			return currhdr;
```

Comment:
A comment here making it explicit that this is the common case where we've found the "real" next proto header would help readability.